confmap/expand_test.go

```diff
@@ -583,21 +583,21 @@ func Test_EscapedEnvVars(t *testing.T) {
 
 	expectedMap := map[string]any{
 		"test_map": map[string]any{
-			"recv.1":  "$MAP_VALUE_1",
-			"recv.2":  "$$MAP_VALUE_2",
-			"recv.3":  "$$MAP_VALUE_3",
-			"recv.4":  "$" + mapValue2,
-			"recv.5":  "some${MAP_VALUE_4}text",
-			"recv.6":  "${ONE}${TWO}",
-			"recv.7":  "text$",
-			"recv.8":  "$",
-			"recv.9":  "${1}${env:2}",
-			"recv.10": "some${env:MAP_VALUE_4}text",
-			"recv.11": "${env:" + mapValue2 + "}",
-			"recv.12": "${env:${MAP_VALUE_2}}",
-			"recv.13": "env:MAP_VALUE_2}${MAP_VALUE_2}{",
-			"recv.14": "${env:MAP_VALUE_2${MAP_VALUE_2}",
-			"recv.15": "$" + mapValue2,
+			"recv.1":  "$$MAP_VALUE_1",
```

Comment:
Can you add tests in `confmap/internal/e2e` that tests all of these scenarios being passed to the expandconverter?

Reply:
Added a test for cases where we don't expect failures. Do we need to keep your test Test_EscapedEnvVars around?

Reply:
I think we can remove this version of the test.  The new test will fail with the expand converter is removed, prompting us to add the escape logic back to Resolve.